File: model/get_food_items.py
from flask import Flask, request
import mysql.connector
from mysql.connector import Error
import html

app = Flask(__name__)

db_config = {
    'host': 'localhost',
    'user': 'root',
    'password': 'ishan@2001',
    'database': 'pfsa'
}

@app.route('/get_food_items', methods=['POST'])
def get_food_items():
    category = request.form.get('category', '')

    if not category:
        return "No category provided", 400

    try:
        connection = mysql.connector.connect(**db_config)
        cursor = connection.cursor(dictionary=True)
        query = "SELECT food_id, food_name FROM food WHERE category = %s ORDER BY food_name"
        cursor.execute(query, (category,))
        results = cursor.fetchall()

        options = '<option value="" disabled selected>Select a food item</option>'
        for row in results:
            food_id = html.escape(str(row['food_id']))
            food_name = html.escape(row['food_name'])
            options += f'<option value="{food_id}">{food_name}</option>'

        return options

    except Error as e:
        return f"Database error: {e}", 500

    finally:
        if connection.is_connected():
            cursor.close()
            connection.close()

if __name__ == '__main__':
    app.run(debug=True)

File: get_food_items.php
<?php
include 'db_connect.php';

if (isset($_POST['category'])) {
    $category = $conn->real_escape_string($_POST['category']);
    $query = "SELECT food_id, food_name FROM food WHERE category = ? ORDER BY food_name";
    $stmt = $conn->prepare($query);
    $stmt->bind_param("s", $category);
    $stmt->execute();
    $result = $stmt->get_result();

    $options = '<option value="" disabled selected>Select a food item</option>';
    while ($row = $result->fetch_assoc()) {
        $options .= '<option value="'.htmlspecialchars($row['food_id']).'">'.htmlspecialchars($row['food_name']).'</option>';
    }
    
    echo $options;
    $stmt->close();
}
$conn->close();
?>